ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionFragment.kt

```diff
@@ -102,6 +101,16 @@ class DataCollectionFragment : AbstractFragment(), BackPressListener {
         }
       }
     )
+
+    lifecycleScope.launch { viewModel.uiState.filterNotNull().collect { updateUI(it) } }
+  }
+
+  private fun updateUI(uiState: UiState) {
+    when (uiState) {
+      is UiState.TaskListAvailable -> loadTasks(uiState.tasks, uiState.taskPosition)
+      is UiState.TaskUpdated -> onTaskChanged(uiState.taskPosition)
+      is UiState.Error -> popups.ErrorPopup().show(uiState.message)
```

Comment:
Showing an error popup is an event, not a state, since it's fired once then forgotten. To avoid the error reappearing in each config change or other lifecycle event, I would model this as a hot non-replaying stream instead (Coroutine).

Reply:
Removed error from UiState